src/bindings/c/ov/src/ov_c_api.cpp

```diff
@@ -0,0 +1,1428 @@
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_node_get_tensor_shape(ov_output_node_list_t* nodes, size_t idx,
+                                    ov_shape_t* tensor_shape) {
+    if (!nodes || idx >= nodes->num) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        auto shape = nodes->output_nodes[idx].object->get_shape();
+        if (shape.size() > MAX_DIMENSION) {
+            return ov_status_e::GENERAL_ERROR;
+        }
+        tensor_shape->ranks = shape.size();
+        std::copy_n(shape.begin(), shape.size(), tensor_shape->dims);
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_node_get_tensor_type(ov_output_node_list_t* nodes, size_t idx,
+                                    ov_element_type_e *tensor_type) {
+    if (!nodes || idx >= nodes->num) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        auto type = (ov::element::Type_t)nodes->output_nodes[idx].object->get_element_type();
+        *tensor_type = (ov_element_type_e)type;
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_model_get_input_by_name(const ov_model_t* model,
+                                const char* tensor_name,
+                                ov_output_node_t **input_node) {
+    if (!model || !tensor_name || !input_node) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+    try {
+        auto result = std::const_pointer_cast<const ov::Model>(model->object)->input(tensor_name);
+        *input_node = new ov_output_node_t;
+        (*input_node)->object = std::make_shared<ov::Output<const ov::Node>>(std::move(result));
+    } CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_model_get_input_by_id(const ov_model_t* model,
+                                const size_t index,
+                                ov_output_node_t **input_node) {
+    if (!model || index < 0 || !input_node) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+    try {
+        auto result = std::const_pointer_cast<const ov::Model>(model->object)->input(index);
+        *input_node = new ov_output_node_t;
+        (*input_node)->object = std::make_shared<ov::Output<const ov::Node>>(std::move(result));
+    } CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+bool ov_model_is_dynamic(const ov_model_t* model) {
+    if (!model) {
+        printf("[ERROR] The model is NULL!!!\n");
+        return false;
+    }
+    return model->object->is_dynamic();
+}
+
+std::vector<char*> split(std::string s) {
+    std::vector<char*> result;
+    std::regex delimiter("[,]");
+    std::sregex_token_iterator tokens(s.cbegin(), s.cend(), delimiter, -1);
+    std::sregex_token_iterator end;
+    for (; tokens != end; ++tokens) {
+        result.push_back(str_to_char_array(*tokens));
+    }
+    return result;
+}
+
+ov_status_e ov_partial_shape_init(ov_partial_shape_t* partial_shape, const char* str) {
```

Comment:
Can we init partial shape from dymensions? Not from string

Reply:
Init a partial shape from string maybe more friendly for user.